### DISCUSSION TITLE
"subplot" title support while using multiple shared axis

### DISCUSSION THREAD

**Discussion by smetis** [oxyplot/oxyplot#1953] [Q&A] [answered]
Hi,
I have a need to show multiple plots (subplots) and find using multiple axis shall meet the requirement except that I cannot find a way to render title per subplot. I believe there is no built-in support on this but will be happy to implement a custom approach to enable this function.

Is there any suggestion where this should be implemented? during rendering of title? axis etc?

**Comment by VisualMelon** [comment, marked as answer]
This is indeed not directly supported: the simplest solution would be to add a custom annotation (i.e. extend `OxyPlot.Annotations.Annotation`) that just renders a piece of text in suitable location (e.g. relative to a specified pair of axes): custom annotations are very easy to implement and enable a composable and easy-to-use solution; modifying existing components is hard work. Using an annotation does, however, mean you'll need to manage the plot margins yourself.

**Reply by smetis** [reply]
Using annotation for subplot title works for me. thanks. 

![image](https://user-images.githubusercontent.com/6990404/199547382-a7f36f6f-20cc-4dec-b796-fa347a9fb276.png)

**Reply by VisualMelon** [reply]
Cool, thanks for letting us know what worked.

**Comment by smetis** [comment]
I have some experience in custom annotation, just not familiar with how multiple axes works and how to get different axes position to render the text. If annotation sounds like a better way to achieve this, I will look into it. Thanks!
